Adds JSON output format

leonelgalan opened the pull request:
Includes both rspec and cucumber tests as seen on the yaml report.
_____

I needed JSON output while setting up a CI with Reek. I was using the YamlReport and `YAML.parse(output).to_ruby` to get a Hash, but I thought this could be useful for others as well.

I'm getting some 1 failure on rspec and 1 on cucumber, but they are not related to the files changed on this PR. I'm getting the same failures when pulling from master. Maybe is something on my local setup, so I thought I'll give this a try.

mvz commented:
Very useful. Looks good to me.

troessner commented on spec/reek/cli/json_report_spec.rb:
```diff
@@ -0,0 +1,23 @@
+require 'reek/cli/report/formatter'
+
+include Reek
+include Reek::Cli
```
I know we are doing this a lot in our codebase but at least for new specs, lets not do the "include Reek" dance and use fully qualified constants.

troessner commented:
Excellent addition!
Lets fix this one thing i commented on and re-squash - then i'll happily merge..;)

chastell commented on lib/reek/cli/report/report.rb:
```diff
@@ -102,6 +103,15 @@ def show
       #
+      # Displays a list of smells in JSON format
+      # JSON with empty array for 0 smells
+      class JsonReport < Base
```
Does it have to be `JsonReport` rather than `JSONReport`?

(The only reason for these capitalisation abominations I heard about was related to Rails’ code autoloading, but I’m working on a Rails 2.3 app at the moment and even it would find `JSONRerport` in `json_report.rb` just fine.)

leonelgalan commented:
Fixed and squashed! Thanks again for building Reek!

@chastell, I feel you but we the same applies to Yaml, Html, Cli, etc. It might require a major refactor to change all this and add consistency across the project.

troessner commented:
Great job, merged, thanks!

chastell commented:
> @chastell, I feel you but we the same applies to Yaml, Html, Cli, etc. It might require a major refactor to change all this and add consistency across the project.

Challenge accepted. :bowtie:
